src/assets/wise5/classroomMonitor/classroomMonitorComponents/shared/grading-helpers/grading-helpers.ts

```diff
@@ -0,0 +1,24 @@
+import { ComponentContent } from '../../../../common/ComponentContent';
+import { ComponentStatus } from '../../../../common/ComponentStatus';
+
+export function calculateComponentVisibility(
+  components: ComponentContent[],
+  componentIdToHasWork: { [componentId: string]: boolean },
+  componentStatuses: { [componentId: string]: ComponentStatus }
+): { [componentId: string]: boolean } {
+  const componentIdToIsVisible = {};
+  for (const component of components) {
+    componentIdToIsVisible[component.id] =
+      componentIdToHasWork[component.id] &&
+      isComponentVisibleToStudent(componentStatuses, component.id);
+  }
```

Comment:
If you use for...in loop to loop through the componentIdToHasWork object, we won't need to pass in the components param to this function?
```suggestion
  for (const componentId in componentIdToHasWork) {
    componentIdToIsVisible[componentId] =
      componentIdToHasWork[componentId] &&
      isComponentVisibleToStudent(componentStatuses, componentId);
  }
```